.github/workflows/ci.yml

```
@@ -1,4 +1,8 @@
-on: push
+on: 
+  push:
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize, reopened]
```

It seems to me we can just run CI on any PR
```suggestion
on: 
  pull_request:
```

---

I think the CI badge is not working properly after removing `on: push` 😓